controller/src/main/java/io/pravega/controller/timeout/TimerWheelTimeoutService.java

```diff
@@ -220,6 +219,10 @@ public PingTxnStatus pingTxn(final String scope, final String stream, final UUID
 
         final TxnData txnData = map.get(key);
 
+        if (txnData == null) {
```

Comment:
I suppose this can only happen in the case we concurrently remove the entry from the timer wheel and ping it. We remove it from timer wheel in two occasions:
1- Abort the txn
2- Change of ownership

I'm trying to assess the consequences of throwing here according to these possible scenarios.

Reply:
we remove it in case of 1 and not 2.
txnwheelservice is not aware of change of ownership.. it only knows about it when it tries to abort the txn at the end of a timeout.

Reply:
ok, so in the case we ping and abort concurrently, and throw here for the ping, is this simply going to cause the ping to fail?

Reply:
yes